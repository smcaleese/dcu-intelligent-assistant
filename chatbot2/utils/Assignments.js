import axios from 'axios';
import { apiURL } from '../config'

export const getAssignments = async (email) => {
    let assignments = []
    try {
        var myAssignmentsData = await axios.post(apiURL + 'getAssignments', {email: email})
        assignments = myAssignmentsData.data
    } catch(err) {
        console.log("error:", err)
    }
    return assignments
}

export const addAssignment = async (emailValue, inputValue, dateValue) => {
    try {
        const postResponse = await axios.post(apiURL + "addAssignment", {
            email: emailValue,
            name: inputValue,
            date: dateValue
        })
        const message = postResponse.data
        return message
    } catch(err) {
        console.log("error:", err)
    }
}


export const deleteAssignment = async (assignment) => {
    const name = assignment["name"]
    try {
        const deleteAssignmentResponse = await axios.post(apiURL + 'deleteAssignment', {name: name})
        const message = deleteAssignmentResponse.data
        return message
    } catch(err) {
        console.log("error:", err)
    }
}

