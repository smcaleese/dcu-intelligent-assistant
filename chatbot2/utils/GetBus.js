import axios from 'axios';
import { apiURL } from '../config'

export const getStops = async () => {
    return new Promise((resolve) => {
        axios.get(apiURL + "getBusStops")
        .then((res) => {
            resolve(res.data)
        })
        .catch(() => console.error("Could not get bus stops"))
    })
}

export const getBusByStop = async (stop_id) => {
    return new Promise((resolve) => {
        axios.post(apiURL + "getStopTimetable", {
            stop_id: stop_id
        })
        .then((res) => {
            resolve(res.data)
        })
        .catch((err) => console.error(err))
    })
}

export const getBusByNumber = async (bus_number) => {
    return new Promise((resolve) => {
        axios.post(apiURL + "getBusTimetable", {
            bus_number: bus_number.toUpperCase()
        })
        .then((res) => {
            resolve(res.data)
        })
        .catch((err) => console.error(err))
    })
}

export const getStopID = async (stop_data) => {
    return new Promise((resolve, reject) => {
        axios.get(apiURL + "getBusStops")
        .then((res) => {
            res.data.stops.forEach((stop) => {
                if (Object.values(stop).findIndex(item => item.toLowerCase() === stop_data.toLowerCase()) !== -1 || Object.values(stop).findIndex(item => item.toLowerCase() === ("stop " + stop_data).toLowerCase()) !== -1){
                    resolve([stop.stop_id, stop.stop_name])
                }
            });
            reject("Error: No stop id could be identified")
        })
        .catch(() => console.error("Could not get bus stops"))
    })
}