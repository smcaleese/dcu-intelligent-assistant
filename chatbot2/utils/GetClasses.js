import axios from 'axios';
import { apiURL } from '../config'

export const getClasses = async (rawCourseCode, date, err='') => {
    return new Promise((resolve, reject) => {
        const courseCode = rawCourseCode.toUpperCase()

        if (err) {
            reject("Lex error")
        }
        else{
            console.log("dataToPost: courseCode:", courseCode, "date:", date, "post URL:", apiURL)

            axios.post(apiURL + 'timetable', {
                courseCode: courseCode,
                date: date
            })
            .then(res => {
                const dataToSend = {
                    ...res.data,
                    courseCode: courseCode
                }
                resolve(dataToSend)
            })
            .catch(err => {
                console.log("Issue with axios POST request to server:", err)
            })
        }
    })
}