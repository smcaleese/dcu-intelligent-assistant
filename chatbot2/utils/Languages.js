export const langs = [
   {
      "lang_name":"Auto",
      "lang_code":"auto"
   },
   {
      "lang_name":"Afrikaans",
      "lang_code":"af"
   },
   {
      "lang_name":"Albanian",
      "lang_code":"sq"
   },
   {
      "lang_name":"Amharic",
      "lang_code":"am"
   },
   {
      "lang_name":"Arabic",
      "lang_code":"ar"
   },
   {
      "lang_name":"Armenian",
      "lang_code":"hy"
   },
   {
      "lang_name":"Azerbaijani",
      "lang_code":"az"
   },
   {
      "lang_name":"Bengali",
      "lang_code":"bn"
   },
   {
      "lang_name":"Bosnian",
      "lang_code":"bs"
   },
   {
      "lang_name":"Bulgarian",
      "lang_code":"bg"
   },
   {
      "lang_name":"Catalan",
      "lang_code":"ca"
   },
   {
      "lang_name":"Chinese (Simplified)",
      "lang_code":"zh"
   },
   {
      "lang_name":"Chinese (Traditional)",
      "lang_code":"zh-TW"
   },
   {
      "lang_name":"Croatian",
      "lang_code":"hr"
   },
   {
      "lang_name":"Czech",
      "lang_code":"cs"
   },
   {
      "lang_name":"Danish",
      "lang_code":"da"
   },
   {
      "lang_name":"Dari",
      "lang_code":"fa-AF"
   },
   {
      "lang_name":"Dutch",
      "lang_code":"nl"
   },
   {
      "lang_name":"English",
      "lang_code":"en"
   },
   {
      "lang_name":"Estonian",
      "lang_code":"et"
   },
   {
      "lang_name":"Farsi (Persian)",
      "lang_code":"fa"
   },
   {
      "lang_name":"Filipino Tagalog",
      "lang_code":"tl"
   },
   {
      "lang_name":"Finnish",
      "lang_code":"fi"
   },
   {
      "lang_name":"French",
      "lang_code":"fr"
   },
   {
      "lang_name":"French (Canada)",
      "lang_code":"fr-CA"
   },
   {
      "lang_name":"Georgian",
      "lang_code":"ka"
   },
   {
      "lang_name":"German",
      "lang_code":"de"
   },
   {
      "lang_name":"Greek",
      "lang_code":"el"
   },
   {
      "lang_name":"Gujarati",
      "lang_code":"gu"
   },
   {
      "lang_name":"Haitian Creole",
      "lang_code":"ht"
   },
   {
      "lang_name":"Hausa",
      "lang_code":"ha"
   },
   {
      "lang_name":"Hebrew",
      "lang_code":"he"
   },
   {
      "lang_name":"Hindi",
      "lang_code":"hi"
   },
   {
      "lang_name":"Hungarian",
      "lang_code":"hu"
   },
   {
      "lang_name":"Icelandic",
      "lang_code":"is"
   },
   {
      "lang_name":"Indonesian",
      "lang_code":"id"
   },
   {
      "lang_name":"Italian",
      "lang_code":"it"
   },
   {
      "lang_name":"Japanese",
      "lang_code":"ja"
   },
   {
      "lang_name":"Kannada",
      "lang_code":"kn"
   },
   {
      "lang_name":"Kazakh",
      "lang_code":"kk"
   },
   {
      "lang_name":"Korean",
      "lang_code":"ko"
   },
   {
      "lang_name":"Latvian",
      "lang_code":"lv"
   },
   {
      "lang_name":"Lithuanian",
      "lang_code":"lt"
   },
   {
      "lang_name":"Macedonian",
      "lang_code":"mk"
   },
   {
      "lang_name":"Malay",
      "lang_code":"ms"
   },
   {
      "lang_name":"Malayalam",
      "lang_code":"ml"
   },
   {
      "lang_name":"Maltese",
      "lang_code":"mt"
   },
   {
      "lang_name":"Mongolian",
      "lang_code":"mn"
   },
   {
      "lang_name":"Norwegian",
      "lang_code":"no"
   },
   {
      "lang_name":"Persian",
      "lang_code":"fa"
   },
   {
      "lang_name":"Pashto",
      "lang_code":"ps"
   },
   {
      "lang_name":"Polish",
      "lang_code":"pl"
   },
   {
      "lang_name":"Portuguese",
      "lang_code":"pt"
   },
   {
      "lang_name":"Romanian",
      "lang_code":"ro"
   },
   {
      "lang_name":"Russian",
      "lang_code":"ru"
   },
   {
      "lang_name":"Serbian",
      "lang_code":"sr"
   },
   {
      "lang_name":"Sinhala",
      "lang_code":"si"
   },
   {
      "lang_name":"Slovak",
      "lang_code":"sk"
   },
   {
      "lang_name":"Slovenian",
      "lang_code":"sl"
   },
   {
      "lang_name":"Somali",
      "lang_code":"so"
   },
   {
      "lang_name":"Spanish",
      "lang_code":"es"
   },
   {
      "lang_name":"Spanish (Mexico)",
      "lang_code":"es-MX"
   },
   {
      "lang_name":"Swahili",
      "lang_code":"sw"
   },
   {
      "lang_name":"Swedish",
      "lang_code":"sv"
   },
   {
      "lang_name":"Tagalog",
      "lang_code":"tl"
   },
   {
      "lang_name":"Tamil",
      "lang_code":"ta"
   },
   {
      "lang_name":"Telugu",
      "lang_code":"te"
   },
   {
      "lang_name":"Thai",
      "lang_code":"th"
   },
   {
      "lang_name":"Turkish",
      "lang_code":"tr"
   },
   {
      "lang_name":"Ukrainian",
      "lang_code":"uk"
   },
   {
      "lang_name":"Urdu",
      "lang_code":"ur"
   },
   {
      "lang_name":"Uzbek",
      "lang_code":"uz"
   },
   {
      "lang_name":"Vietnamese",
      "lang_code":"vi"
   },
   {
      "lang_name":"Welsh",
      "lang_code":"cy"
   }
]