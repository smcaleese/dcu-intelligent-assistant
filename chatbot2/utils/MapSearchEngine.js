
export const mapSearchEngine = (input, mapData) => {
    const inputAllCaps = input.toUpperCase();
    const inputAllCapsArr = inputAllCaps.split(' '); // eg. ["HENRY", "GRATTAN", "BUILDING"]

    let searchTable = [];

    for(let i = 0; i < mapData.length; i++) {
        const currentTitle = mapData[i]["properties"].title
        const currentCoords = mapData[i]["geometry"].coordinates
        const currentTitleArr = currentTitle.split(' ');
        searchTable[i] = [currentTitle, currentCoords, 0]

        // count the number of word matches between the input string and each entry in the JSON database
        for(let j = 0; j < inputAllCapsArr.length; j++) {
            for(let k = 0; k < currentTitleArr.length; k++) {
                if(inputAllCapsArr[j] === currentTitleArr[k].toUpperCase()) {
                    searchTable[i][2] += 1
                }
            }
        }
    }

    const searchTableSorted = searchTable.sort((a, b) => {
        return b[2] - a[2]
    })

    // pick top element from sorted table, the element with the most matches
    const searchStringArr = searchTableSorted[0][0].split(' ')
    const searchStringResult = searchStringArr.join(' ')

    const searchCoords = searchTableSorted[0][1]
    const finalResult = [searchStringResult, searchCoords]
    return finalResult;
}



