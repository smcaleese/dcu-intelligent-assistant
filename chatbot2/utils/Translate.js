import AWS from 'aws-sdk/dist/aws-sdk-react-native';
import { awsRegion, awsIdentityPool } from '../config'

AWS.config.region = awsRegion
AWS.config.credentials = new AWS.CognitoIdentityCredentials({
    IdentityPoolId: awsIdentityPool
})

const translateRunner = new AWS.Translate();

export const TranslateText = (text, targetLang='en', soureLang='auto') => {
    return new Promise((resolve, reject) => {
        const params = {
            SourceLanguageCode: soureLang,
            TargetLanguageCode: targetLang,
            Text: text
        };
        translateRunner.translateText(params, (err, data) => {
            if (err){
                reject(err)
            }
            else {
                resolve(data);
            }
        })
    })
}
