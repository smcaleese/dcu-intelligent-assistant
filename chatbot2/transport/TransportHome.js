import React, { useState } from 'react';
import { StyleSheet, View, Text, TextInput } from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { getBusByStop, getBusByNumber, getStops } from '../utils/GetBus';
import CustomButton from '../components/CustomButton';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#f0f0f0',
        alignItems: 'center'
    },
    titleContainer: {
        backgroundColor: "#4269E1",
        width: "80%",
        borderTopLeftRadius: 10,
        borderTopRightRadius: 10,
        padding: 5,
        marginTop: 50
    },
    titleText: {
        textAlign: 'center',
        fontSize: 20,
        color: "white",
        padding: 5
    },
    form: {
        backgroundColor: 'white',
        borderBottomLeftRadius: 10,
        borderBottomRightRadius: 10,
        paddingTop: 30,
        width: '80%',
        justifyContent: 'center',
        alignItems: 'center'
    },
    dropDownContainer: {
        height: 50, 
        width: '80%'
    },  
    dropDown: {
        backgroundColor: '#f8f8f8'
    },
    dropDownText: {
        fontSize: 16
    },
    dropDownItem: {
        justifyContent: 'flex-start'
    },
    dropDownLabel: {
        padding: 2
    },
    busNumInput: {
        borderWidth: 1,
        borderColor: "#c3c3c3",
        borderRadius: 5,
        paddingLeft: 20,
        width: '80%',
        height: 50,
        fontSize: 16
    },
    customButtonView: {
        margin: 16
    }
})

export default function TransportHome(props) {

    //Gets a list of all stops around DCU from server. Used for the dropdown items
    const getStopData = () => {
        getStops()
        .then((res) => {
            busStops = []
            res.stops.forEach((stop) => {
                busStops.push({label: stop.stop_name + (stop.stop_number.replace('stop ', ' ')), value: stop.stop_id, icon: () => <Icon name="bus-stop" size={18} color="#900" />})
                setBusStopState(busStops)
            })
        })
        .catch((err) => (console.log(err)))
    }

    const [busStopState, setBusStopState] = useState(getStopData)
    const [dropdownState, setDropDownState] = useState("")
    const [busNumState, setBusNumState] = useState("")
    //Dropdrown controller for managing the dropdown value
    let controller;

    const handleDropDownChange = (item) =>{
        //If the drop down is cleared do not reset the text input
        if (!item.value){
            setDropDownState(item.value);
        }
        else if (item.value !== dropdownState){
            setDropDownState(item.value);
            setBusNumState("");
        }
    }

    const handleSubmit = () => {
        if (busNumState){
            //Make API call to server to get times from bus number
            getBusByNumber(busNumState.toUpperCase())
            .then((res) => {
                props.navigation.navigate('output', { times: res, input: busNumState, type: "bus" })
            })
            .catch((err) => console.log(err))
        }
        else if (dropdownState){
            //Make API call to server to get times from bus stop
            getBusByStop(dropdownState)
            .then((res) => {
                props.navigation.navigate('output', { times: res, input: dropdownState, type: "stop" })
            })
            .catch((err) => console.log(err))
        }
    }

    return (
        <View style={styles.container}>
            <View style={styles.titleContainer}>
                <Text style={styles.titleText}>Bus Information</Text>
            </View>
            <View style={styles.form}>
                <DropDownPicker
                    items={busStopState}
                    defaultValue={dropdownState}
                    controller={instance => controller = instance}
                    dropDownMaxHeight={190}
                    placeholder="Select a Stop"
                    containerStyle={styles.dropDownContainer}
                    style={styles.dropDown}
                    globalTextStyle={styles.dropDownText}
                    itemStyle={styles.dropDownItem}
                    labelStyle={styles.dropDownLabel}
                    onChangeItem={item => {handleDropDownChange(item)}}
                /> 
                <Text style={{fontSize: 16, padding: 10}}> or </Text>
                <TextInput
                    style={styles.busNumInput}
                    placeholder={"Bus Number"}
                    value={busNumState}
                    onChangeText={(val) => {setBusNumState(val); controller.reset()}}
                />
                <View style={styles.customButtonView}>
                    <CustomButton title="Submit" color="green" onPress={handleSubmit}/>
                </View>
            </View>
        </View>
    )
}