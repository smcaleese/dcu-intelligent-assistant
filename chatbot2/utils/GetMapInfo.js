import { apiURL } from '../config';
import axios from 'axios';
import { mapSearchEngine } from './MapSearchEngine';

export const getMapData = async () => {
    try {
        const res = await axios.get(apiURL + 'mapData')
        return res.data
    } 
    catch(err) {
        console.log("error:", err)
    }
}

export const searchMap = async (location) => {
    try {
        const mapData = await getMapData()
        const [title, coords] = mapSearchEngine(location, mapData.locations)
        return { title, coords }
    }
    catch(err) {
        console.log("error:", err)
    }
}