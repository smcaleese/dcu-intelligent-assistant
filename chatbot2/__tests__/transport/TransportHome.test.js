import React from 'react';
import {create} from 'react-test-renderer';
import TransportHome from '../../transport/TransportHome';
import { getStops } from '../../utils/GetBus';

jest.mock('../../utils/GetBus', () => ({ getStops: jest.fn() }))
jest.useFakeTimers();

getStops.mockImplementation(() => {
    return new Promise((resolve) => {
        resolve([])
    })
})

const screen = create(<TransportHome />)

test('TransportHome Snapshot Test', () => {
    expect(screen).toMatchSnapshot();
})
