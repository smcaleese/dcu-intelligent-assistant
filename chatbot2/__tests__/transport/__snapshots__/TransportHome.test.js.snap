// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransportHome Snapshot Test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f0f0f0",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4269E1",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "marginTop": 50,
        "padding": 5,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      Bus Information
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "justifyContent": "center",
        "paddingTop": 30,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 50,
            "width": "80%",
          },
          Object {
            "zIndex": 5000,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#f8f8f8",
            "borderBottomLeftRadius": 5,
            "borderBottomRightRadius": 5,
            "borderColor": "#dfdfdf",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "borderWidth": 1,
            "flex": 1,
            "flexDirection": "row",
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "fontSize": 16,
              },
              Object {
                "flex": 1,
                "opacity": 1,
              },
              Object {
                "marginLeft": 5,
                "marginRight": 0,
              },
              [Function],
              Object {
                "padding": 2,
              },
              false,
              Object {},
            ]
          }
        >
          Select a Stop
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomLeftRadius": 0,
                "borderTopLeftRadius": 0,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingVertical": 8,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {},
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 15,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Feather",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï„®
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomLeftRadius": 5,
              "borderBottomRightRadius": 5,
              "borderColor": "#dfdfdf",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "borderWidth": 1,
              "paddingHorizontal": 10,
              "paddingVertical": 5,
            },
            Object {
              "alignItems": "center",
              "borderTopLeftRadius": 0,
              "borderTopRightRadius": 0,
              "justifyContent": "center",
              "position": "absolute",
              "width": "100%",
            },
            Object {},
            Object {
              "borderWidth": 0,
              "display": "none",
              "position": "relative",
            },
            Object {
              "maxHeight": 190,
              "top": undefined,
              "zIndex": 5000,
            },
          ]
        }
      >
        <RCTScrollView
          nestedScrollEnabled={true}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "marginBottom": 15,
                  "marginVertical": 10,
                }
              }
            >
              <Text>
                Not Found
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 16,
          "padding": 10,
        }
      }
    >
       or 
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Bus Number"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#c3c3c3",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 16,
          "height": 50,
          "paddingLeft": 20,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "margin": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "green",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "padding": 15,
            "width": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
